lib/src/json_schema/json_schema.dart

```diff
@@ -630,9 +630,11 @@ class JsonSchema {
             }
           }
 
-          // If currentSchema is a ref, resolve ref recursively.
+          // If currentSchema is a ref, we might want to resolve it recursively right now.
+          // If
```

Comment:
Did you have an additional thought?

Reply:
Yes. I do. That's the point I realized the path might not be in the reference. There might be some ambiguity going on....